doc/user/content/transform-data/temporal-filters.md

```diff
@@ -0,0 +1,368 @@
+
+```sql
+-- Reduce down to counts within a time interval.
+SELECT content, count(*)
+FROM events
+WHERE mz_now() >= insert_ts
+  AND mz_now() < delete_ts
+GROUP BY content;
+```
+
+This query will change its results over time. Not just because you might add to and remove from `events`, but because `mz_now()` advances as you stare at your screen. Each time you issue the query you may get a different result.
+
+This looks like a great query! What's not to like?
+
+The main issue really is that it is ***just a query***. You can ask this question over and over, but you can also ask the same thing with an arbitrary timestamp in place of `mz_now()`. To support that, Materialize has to keep the entire collection of data around. Your `events` table will grow and grow and grow, and the time to answer the query will grow as well.
+
+## Time-Windowed Computation
+
+Materialize specializes at ***maintaining*** computations like the above, both because that can make it faster to get your answers out, but also because by specifying what you actually need Materialize can run much more lean. We'll see that now with the query above! Moreover, comparing parts of your data to `mz_now()` in views (rather than just in queries) introduces powerful new idioms, ones that we'll explore in this article.
+
+```sql
+-- Maintained collection of only valid results.
+CREATE MATERIALIZED VIEW valid_events
+AS SELECT content, count(*)
+FROM events
+WHERE mz_now() >= insert_ts
+  AND mz_now() < delete_ts
+GROUP BY content;
+```
+
+What's all this then?
+
+Presumably `valid_events` has the property that if you `SELECT` from it you should see the same results as for the time-windowed `SELECT` in the previous section. That is 100% true.
+
+What is also true is that `valid_events` has enough information from you, in the form of the query itself, to maintain only enough historical detail to answer these `SELECT` queries from now onward. Once Materialize's `mz_now()` passes a record's `delete_ts` it cannot be seen (at least not through this view), and Materialize can dispose of the event. The in-memory footprint of `valid_events` stays bounded by the number of records in the system that could still satisfy this constraint (those records that are currently valid, or who may yet become valid in the future).
+
+While you add to `events`, Materialize collects up the events that are no longer visible, automatically. Of course, you can also ***change*** the records in `events`, in case you want to remove some events early, or draw out the `delete_ts` of any record, or replace one event with its next stage (and new `insert_ts` and `delete_ts`). If you happen to adjust any fields that interact with `mz_now()` Materialize will update the views appropriately.
```

Comment:
Nit, and my fault, but I now would say we "update the results of the view" more than "update the view".